services/spar/src/Spar/Error.hs

```diff
@@ -76,8 +78,18 @@ data SparCustomError
   | SparProvisioningTokenLimitReached
   deriving (Eq, Show)
 
+sparToServantErrIO :: MonadIO m => Log.Logger -> SparError -> m ServantErr
```

Comment:
When do we want to use `sparToServantErrIO` vs `sparToServantErr`? Adding the `IO` to the name says it needs IO but not what for.

Do we always want to log or not? We should probably only keep one of these and deprecate the other; or just go through the trouble of switching them all.

When is one used vs the other?

Reply:
please suggest a better name if you have one; i don't.

the non-IO variant is not exported from the module, i just don't want to inline them in the code, since they are clearly doing two different things on top of each other.

so...  is there anything i should do?

Reply:
In that case please name the exported one `sparToServantErr` and the module-only one something like `sparToServantErr'` or `sparToServantErrHelper; If it's module-local I care slightly less about the name, but in the exported version having the `IO` implies to me that there's a non-IO version available and brings into question what the IO is doing 😄. E.g.; optimize the naming so the exported one is clearest since it's furthest from the source code 👍 

Same with the `renderSparError` ones plz

Reply:
i renamed it according to your suggestion, can we please leave it at that?